test/e2e/aws_test.go

```diff
@@ -1130,12 +1210,17 @@ func makeAWSMachineTemplate(namespace, name, instanceType string, az, subnetId *
 }
 
 func deployCNI(kubeConfigPath, manifestPath string) {
-	Expect(exec.Command(
-		*kubectlBinary,
-		"create",
-		"--kubeconfig="+kubeConfigPath,
-		"-f", manifestPath,
-	).Run()).To(Succeed())
+	Eventually(
+		func() error {
+			return exec.Command(
+				*kubectlBinary,
+				"create",
+				"--kubeconfig="+kubeConfigPath,
+				"-f", manifestPath,
+			).Run()
+		},
+		5*time.Minute, 10*time.Second,
+	).Should(BeNil())
```

Comment:
```suggestion
	).Should(Succeed())
```

Reply:
I'm a bit curious as the why we would need to wrap this with an Eventually.

Reply:
I have observed many failures in the test-grid while trying to deploy CNI. It could be because the DNS for the API server may not be available.